src/core/jsproxy.c

```diff
@@ -751,6 +751,270 @@ JsArray_inplace_concat(PyObject* self, PyObject* other)
+  return result;
+}
+
+static PyMethodDef JsArray_reversed_MethodDef = {
+  "__reversed__",
+  (PyCFunction)JsArray_reversed,
+  METH_NOARGS,
+};
+
+// clang-format off
+EM_JS_NUM(int,
+JsArray_index_helper,
+(JsRef list, JsRef value, int start, int stop),
+{
+  let o = Hiwire.get_value(list);
+  let v = Hiwire.get_value(value);
+  for (let i = start; i < stop; i++) {
+    if (o[i] === v) {
+      return i;
+    }
+  }
+  return -1;
+})
+// clang-format on
+
+PyObject*
+JsArray_index(PyObject* o, PyObject* args)
+{
+  JsProxy* self = (JsProxy*)o;
+  PyObject* value;
+  Py_ssize_t start = 0;
+  Py_ssize_t stop = PY_SSIZE_T_MAX;
+  if (!PyArg_ParseTuple(args, "O|nn:index", &value, &start, &stop)) {
+    return NULL;
+  }
+
+  int length = JsProxy_length(o);
+  if (length == -1) {
+    return NULL;
+  }
+  if (start < 0) {
+    start += length;
+    if (start < 0)
+      start = 0;
+  }
+  if (stop < 0) {
+    stop += length;
+    if (stop < 0)
+      stop = 0;
+  }
+  if (stop > length) {
+    stop = length;
+  }
+
+  JsRef jsvalue = python2js_track_proxies(value, NULL);
+  if (jsvalue == NULL) {
+    PyErr_Clear();
+    for (int i = start; i < stop; i++) {
+      JsRef jsobj = JsArray_Get(self->js, i);
+      PyObject* pyobj = js2python(jsobj);
+      int cmp = PyObject_RichCompareBool(pyobj, value, Py_EQ);
+      hiwire_decref(jsobj);
+      if (cmp > 0)
+        return PyLong_FromSsize_t(i);
+      else if (cmp < 0)
+        return NULL;
+    }
+  } else {
+    int result = JsArray_index_helper(self->js, jsvalue, start, stop);
+    hiwire_decref(jsvalue);
+    if (result != -1) {
+      return PyLong_FromSsize_t(result);
+    }
+  }
+  if (!PyErr_Occurred()) {
+    PyErr_Format(PyExc_ValueError, "%R is not in list", value);
+  }
+  return NULL;
+}
+
+static PyMethodDef JsArray_index_MethodDef = {
+  "index",
+  (PyCFunction)JsArray_index,
+  METH_VARARGS,
+};
+
+// clang-format off
+EM_JS_NUM(int,
+JsArray_count_helper,
+(JsRef list, JsRef value),
+{
+  let o = Hiwire.get_value(list);
+  let v = Hiwire.get_value(value);
+  let result = 0;
+  for (let i = 0; i < o.length; i++) {
+    if (o[i] === v) {
+      result++;
+    }
+  }
+  return result;
+})
+// clang-format on
+
+PyObject*
+JsArray_count(PyObject* o, PyObject* value)
+{
+  JsProxy* self = (JsProxy*)o;
+  JsRef jsvalue = python2js_track_proxies(value, NULL);
+  if (jsvalue == NULL) {
+    PyErr_Clear();
+    int result = 0;
+    Py_ssize_t stop = JsProxy_length(o);
+    if (stop == -1) {
+      return NULL;
+    }
+    for (int i = 0; i < stop; i++) {
+      JsRef jsobj = JsArray_Get(self->js, i);
+      PyObject* pyobj = js2python(jsobj);
+      int cmp = PyObject_RichCompareBool(pyobj, value, Py_EQ);
+      hiwire_decref(jsobj);
+      if (cmp > 0)
+        result++;
+      else if (cmp < 0)
+        return NULL;
+    }
+    return PyLong_FromSsize_t(result);
+  } else {
+    int result = JsArray_count_helper(self->js, jsvalue);
+    hiwire_decref(jsvalue);
+    if (result == -1) {
```

Comment:
Unlikely. If somehow an exception occurs in `JsArray_count_helper` then it would be `-1`. Currently `JsArray_count_helper` probably can't throw an exception. Checking for this case will make it much easier to debug if it is ever modified in a way that does throw.